python/ray/dashboard/modules/job/tests/test_job_manager.py

```diff
@@ -46,6 +48,59 @@
 import psutil
 
 
+@pytest.mark.asyncio
+@pytest.mark.parametrize(
+    "call_ray_start",
+    [
+        {
+            "env": {
+                "RAY_enable_export_api_write": "true",
+            },
+            "cmd": "ray start --head",
+        }
+    ],
+    indirect=True,
+)
+async def test_submission_job_export_events(call_ray_start, tmp_path):  # noqa: F811
+    """
+    Test submission job events are correctly generated and written to file
+    as the job goes through various state changes in its lifecycle.
+    """
+    ray_constants.RAY_ENABLE_EXPORT_API_WRITE = True
```

Comment:
this screws up the global variable for the rest of the tests. you should probably create a fixture if you want to do this. (or alternatively, just update os.environ)

Reply:
Updated to set the environment variable in addition to passing the env var when calling `ray start`, because the constant needs to be set outside ray tasks also when submitting a job.